### DISCUSSION TITLE
(baseball) Drill down into matchup point totals?

### DISCUSSION THREAD

**Discussion by msmith1695** [cwendt94/espn-api#453] [ESPN API]
I'm attempting to use this package to automate league analysis I currently do manually in a spreadsheet. Each week, I enter league point totals from the week's matchup including total points, points from batters, and points from pitchers, and use the data to calculate averages, rolling ranks, etc.

This package is a fantastic resource and I've had a lot of fun diving into it. However, it looks like the level of detail in the `Matchup` object tops out at the total score level. It would be awesome to be able to drill down one step further and get total points from batters and total points from pitchers. And to take it one step further, drilling down into the number of points each individual player contributed to the total would be stellar.

If any of this is possible with existing functionality I haven't discovered yet, I'd appreciate the info!

**Comment by msmith1695** [comment]
Adding - while iterating through the players on a `Team.Roster`, I am finding that all of the `stats` variables on any `Player` are just empty dictionaries - am I using this wrong, or is there functionality still missing from the baseball functions?

**Comment by cwendt94** [comment]
This is currently not possible with the existing baseball functionality but shouldn't be too hard to add. We have this functionality for football and basketball through the [Box Score](https://github.com/cwendt94/espn-api/wiki/Box-Score-Class) and [Box Score Player](https://github.com/cwendt94/espn-api/wiki/Box-Player-Class) that drills down by week with all of your roster/stat information. 

Baseball has not gotten as much updates as the rest but I can look into adding this in the next week. Most of the code and logic from the other sports apply to get some of that basic functionality you are looking for.

**Reply by msmith1695** [reply]
Fantastic! I'll look forward to it!

**Reply by cwendt94** [reply]
Here is the first pass at adding Box Player to Box Score, [PR](https://github.com/cwendt94/espn-api/pull/457). I also updated the regular player that now should have `stats` field filled out. It breaks down the stats by scoring period and then sets two new fields for player `total_points` and `projected_total_points` for that season.
